### DISCUSSION TITLE
NEW ENDPOINT (SHARE PROFILE) FOUND - TESTED - WORKING

### DISCUSSION THREAD

**Discussion by Amethxst** [isFakeAccount/psnawp#88] [Show and tell]
For the past few weeks, it has been possible to share PSN profiles from PS APP (by generating a link and a QR CODE).
I experimented a bit and was able to do everything:

*note: having problems testing (using test files) I directly edited the user.py and endpoints.py files by running the API locally.

![API_1](https://github.com/user-attachments/assets/8bd7abe6-f484-4bf2-bf24-09654714aa41)

![API_2](https://github.com/user-attachments/assets/ddd80228-3af1-43f5-9372-fda8d31e78cc)

**endpoints.py**
- Add new BASE_PATCH:
```
"cpss": "https://m.np.playstation.com/api/cpss"
```

- Add new API_PATH:
```
"share_profile": "/v1/share/profile/{account_id}",
```

**user.py**
- Added 'get_shareable_profile_link'

```
def get_shareable_profile_link(self) -> dict[str, Any]:
        try:
            url = f"{BASE_PATH['cpss']}{API_PATH['share_profile'].format(account_id=self.account_id)}"

            print("URL:", url)  # Debug: api url
        
            response = self.authenticator.get(url=url)
            print("Status Code:", response.status_code)  # Debug: print status
            print("Response Content:", response.content)  # Debug: print response
        
            response_json = response.json()
            return {
                "shareUrl": response_json["shareUrl"],
                "shareImageUrl": response_json["shareImageUrl"]
            }
        except PSNAWPForbidden as forbidden:
            raise PSNAWPForbidden("You are not allowed to access the shareable link of this user.") from forbidden
        except PSNAWPNotFound as not_found:
            raise PSNAWPNotFound(f"Account ID {self.account_id} does not exist.") from not_found
        except Exception as e:
            print("Error API:", e)  # Debug: print any error
            raise e
```

Small script test:

```
import sys
import os

sys.path.insert(0, os.path.abspath("C:\\Users\\meazz\\Desktop\\PSN_APP_RE\\.TEST\\psnawp-master\\src"))

from psnawp_api import PSNAWP
from psnawp_api.models.user import User # type: ignore

def main():

    npsso = "NPSSO"

    psnawp = PSNAWP(npsso)
    online_id = input("ID ONLINE: ")

    try:
        user_example = psnawp.user(online_id=online_id)
        account_id = user_example.account_id
        print(f"\nAccount ID for Online ID '{online_id}': {account_id}")
    except Exception as e:
        print(f"\nErrore Account ID '{online_id}': {e}")
        sys.exit(1)

    try:
        share_info = user_example.get_shareable_profile_link()
        print("\n--- Info ---")
        print(f"Share URL: {share_info['shareUrl']}")
        print(f"Share Image URL: {share_info['shareImageUrl']}")
    except Exception as e:
        print(f"\nError API: {e}")

if __name__ == "__main__":
    main()
```

![API_VIDEO_1](https://github.com/user-attachments/assets/f9758b29-9c8f-4a2a-aa81-963f6c97d7b9)

**Comment by seanzhang98** [comment]
Is there a specific version of the app required? I keep getting 'failed to load' errors when I try to intercept the connection.

**Reply by Amethxst** [reply]
I think the problem occurs because you use AVD

A few weeks ago my AVD completely stopped wanting to start PS APP, while my MAC runs it completely. Maybe they corrected/wrong something in the release of the latest builds

**Reply by isFakeAccount** [reply]
Did you read `CONTRIBUTING.md`? Seems like you didn't inject certificate in the PS App. If you don't do that PS App will know something is monitoring its traffic. Luckily HTTPToolkit does all that as long as you have rooted android phone. 

Another issue is that AVDs on x86_64 machines are well x86_64 so not all AVD's and Android version work. I haven't been able to make anything besides android 11 x86 work. If you have a physical android devices that can be rooted then it might be the easiest way.

**Reply by Amethxst** [reply]
If it helps, it can all be done on iPhone JB as well (always using HTTPToolkit or Burp + related certificates)

But yes, on AVD there are problems in recent builds on Windows (which doesn't happen on macOS)

**Comment by isFakeAccount** [comment]
For the record this endpoint has been added already https://github.com/isFakeAccount/psnawp/pull/96
